pkg/controller/awsfederatedaccountaccess/awsfederatedaccountaccess_controller.go

```diff
@@ -558,22 +558,51 @@ func (r *ReconcileAWSFederatedAccountAccess) removeFinalizer(reqLogger logr.Logg
 func (r *ReconcileAWSFederatedAccountAccess) cleanFederatedRoles(reqLogger logr.Logger, currentFAA *awsv1alpha1.AWSFederatedAccountAccess, federatedRoleCR *awsv1alpha1.AWSFederatedRole) error {
 
 	// Get the UID
-	uidLabel, ok := currentFAA.Labels["uid"]
+	uidLabel, ok := currentFAA.Labels[awsv1alpha1.UIDLabel]
 	if !ok {
 		labelErr := fmt.Sprintf("Unable to get UID label")
 		return errors.New(labelErr)
 	}
 
+	// Get the AWS Account ID
+	accountIDLabel, ok := currentFAA.Labels[awsv1alpha1.AccountIDLabel]
+	if !ok {
+		labelErr := fmt.Sprintf("Unable to get AWS Account ID label")
+		return errors.New(labelErr)
+	}
+
 	roleName := currentFAA.Spec.AWSFederatedRole.Name + "-" + uidLabel
 
-	// Build AWS client
-	awsClient, err := awsclient.GetAWSClient(r.client, awsclient.NewAwsClientInput{
-		SecretName: currentFAA.Spec.AWSCustomerCredentialSecret.Name,
-		NameSpace:  currentFAA.Spec.AWSCustomerCredentialSecret.Namespace,
+	// Build AWS client from root secret
+	rootAwsClient, err := awsclient.GetAWSClient(r.client, awsclient.NewAwsClientInput{
```

Comment:
This doesn't affect @ArielLima's work on the awsclient refactor, correct?  Since it's a different client with different credentials?

Reply:
This will need to be changed after the AWS Client refactor PR is merged I think